omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/TaskDefinitionController1_8Test.java

```diff
@@ -0,0 +1,165 @@
+	}
+	
+	@Test
+	public void shouldGetTaskByUuid() throws Exception {
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI() + "/" + getUuid());
+		SimpleObject result = deserialize(handle(req));
+		Assert.assertNotNull(result.get("name"));
+		Assert.assertNotNull(result.get("description"));
+		Assert.assertNotNull(result.get("taskClass"));
+	}
+	
+	@Test
+	@Override
+	public void shouldGetFullByUuid() throws Exception {
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI() + "/" + getUuid());
+		req.addParameter("v", "full");
+		
+		SimpleObject result = deserialize(handle(req));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "name"));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "description"));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "taskClass"));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "repeatInterval"));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "startOnStartup"));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "started"));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "properties"));
+	}
+	
+	@Test
+	@Override
+	public void shouldGetDefaultByUuid() throws Exception {
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI() + "/" + getUuid());
+		SimpleObject result = deserialize(handle(req));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "name"));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "description"));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "taskClass"));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "startTime"));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "started"));
+	}
+	
+	@Test
+	@Override
+	public void shouldGetRefByUuid() throws Exception {
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI() + "/" + getUuid());
+		req.addParameter("v", "ref");
+		SimpleObject result = deserialize(handle(req));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "name"));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "description"));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "taskClass"));
+	}
+	
+	@Test
+	public void shouldSaveTaskDefinition() throws Exception {
+		SimpleObject mockTask = new SimpleObject();
+		mockTask.add("name", "MockTask");
+		mockTask.add("description", "MockTask Description");
+		mockTask.add("taskClass", "org.openmrs.scheduler.tasks.TestTask");
+		mockTask.add("startTime", "2017-08-28T23:59:59.000+0530");
+		mockTask.add("repeatInterval", "10");
+		mockTask.add("startOnStartup", false);
+		mockTask.add("properties", null);
+		String json = new ObjectMapper().writeValueAsString(mockTask);
+		
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		deserialize(handle(req));
+		Assert.assertEquals(mockTaskServiceWrapper.getTaskByName("MockTask").getName(), "MockTask");
```

Comment:
How sure can i be that this task did not exist before saving?

Reply:
I will add another test before to this to confirm that